core/embed/rust/src/ui/layout_samson/component/button_controller.rs

```diff
@@ -469,6 +476,20 @@ impl Component for ButtonController {
                                     Some(ButtonControllerMsg::Pressed(ButtonPos::Middle)),
                                 )
                             } else {
+                                // When a button gets pressed while another one is being held,
+                                // cancel the hold and do not register the press.
+                                match which_down {
+                                    PhysicalButton::Left => {
+                                        self.left_btn.hold_canceled(ctx);
+                                        return None;
+                                    }
+                                    PhysicalButton::Right => {
+                                        self.right_btn.hold_canceled(ctx);
+                                        return None;
+                                    }
+                                    _ => {}
```

Comment:
Because the power button does not exist in t2b1 nor t3b1 models, I would suggest returning None even in this case and  remove returning of  the `ButtonState::BothDown`
```suggestion
                                    _ => {
                                        return None;
                                    }
```

Reply:
OK, just returned None in all cases.